xarray/backends/zarr.py

```diff
@@ -751,7 +751,7 @@ def set_variables(self, variables, check_encoding_set, writer, unlimited_dims=No
             writer.add(v.data, zarr_array, region)
 
     def close(self):
-        pass
+        self.zarr_group.store.close()
```

Comment:
We likely want to do this here only when the Zarr backend was responsible for opening the ZipStore. You may need to add some logic/state in `open_group` to manage this.

Reply:
That seems reasonable, yes. I added some logic for it and a test case that would have failed before this change.